pytorch_lightning/utilities/distributed.py

```diff
@@ -376,3 +377,32 @@ def init_dist_connection(
             f"All distributed processes registered. Starting with {world_size} processes\n"
             f"{'-' * 100}\n"
         )
+
+
+def _collect_states_on_rank_zero(state: Dict[str, Any], device: torch.device):
+    """This distributed utility collects dictionary state across all processes.
+
+    Args:
+        state: Dictionary contain the state of the current process
+        device: Current process device.
+
+    Returns:
+        states: On global rank 0, a dictionary where the primary keys are
+            the process rank and the values their associated states. Otherwise, returns None.
+    """
+    if not distributed_available():
+        return {0: state}
+    states = {}
+    state = apply_to_collection(state, torch.Tensor, lambda x: x.to(device))
+    for rank in range(1, torch.distributed.get_world_size()):
+        if torch.distributed.get_rank() == rank:
+            # Assumes world_size of 3.
+            objects = [state]
+        else:
+            objects = [None]
+        torch.distributed.broadcast_object_list(objects, src=rank, device=device)
```

Comment:
Should we maybe move it to cpu before syncing to avoid gpu out of memory?

Reply:
No, you need everything on GPU.

Reply:
<img width="813" alt="Screenshot 2021-11-19 at 14 04 09" src="https://user-images.githubusercontent.com/12861981/142635315-f50629aa-0398-47c9-9e9e-371b384c4a4f.png">

Reply:
why? It is moved to cpu at the end anyway.

Reply:
You always need to move the tensors on device before doing a collective operations.

Reply:
It works without it actually. Removed !